include/gsl/byte

```diff
@@ -72,6 +74,12 @@
 #define byte_may_alias
 #endif // defined __clang__ || defined __GNUC__
 
+#if GSL_USE_STD_BYTE
+#define BYTE_TYPE std::byte
+#else // !GSL_USE_STD_BYTE
+#define BYTE_TYPE byte
+#endif // GSL_USE_STD_BYTE
+
```

Comment:
Why is that macro introduced? It was not needed before. Did some code not compile without this macro?

And why is this macro exported?

A macro with a generic name like `BYTE_TYPE` looks like it has potential to break existing code.